types/three/three-core.d.ts

```diff
@@ -836,6 +836,7 @@ export class BufferGeometry extends EventDispatcher {
 
     getIndex(): BufferAttribute;
     setIndex( index: BufferAttribute ): void;
+	setIndex( index: any[] ): void;
```

Comment:
We can do better than `any[]` here. When an array is passed to setIndex, it ultimately gets passed to a TypedArray constructor see:
https://github.com/mrdoob/three.js/blob/master/src/core/BufferGeometry.js#L58
https://github.com/mrdoob/three.js/blob/35a26f178c523514673d992da1aece23c1cfca6e/src/core/BufferAttribute.js#L387

The official signature for those is basically `ArrayLike<number> | ArrayBufferLike`, see:
https://github.com/Microsoft/TypeScript/blob/cd2ea9a12f2bf42fcd4747782b34543a6c36c08c/lib/lib.es6.d.ts#L2035